Lib/test/test_cmd_line.py

```diff
@@ -972,10 +972,32 @@ def test_python_legacy_windows_fs_encoding(self):
 
     @unittest.skipUnless(support.MS_WINDOWS, 'Test only applicable on Windows')
     def test_python_legacy_windows_stdio(self):
-        code = "import sys; print(sys.stdin.encoding, sys.stdout.encoding)"
-        expected = 'cp'
-        rc, out, err = assert_python_ok('-c', code, PYTHONLEGACYWINDOWSSTDIO='1')
-        self.assertIn(expected.encode(), out)
+        fn = os_helper.TESTFN
+        # We cannot use PIPE to test ConsoleIO
+        code = dedent(f"""
+        import sys
+        with open({fn!r}, 'w', encoding='utf-8') as f:
+            print(type(sys.stdout.buffer.raw).__name__, file=f)
+        """)
```

Comment:
```suggestion
        code = "import sys; sys.exit(type(sys.stdout.buffer.raw).__name__ != '_WindowsConsoleIO')"
```

Perhaps? And similar for the second check.